### PR TITLE
[10-10CG] download_pdf refactor

### PR DESCRIPTION
**Pull request by coope93**
## Summary

- This work is behind a feature toggle (flipper): YES
- Replace directly calling `PdfFill::Filler.fill_form` and instead call the `to_pdf` method on the claim instance (which calls `PdfFill::Filler.fill_form` in the method but also has additional logging.
- This change simplifies the pdf generation into the `to_pdf` method and ensures a consistent place to validate and log the pdf generation logic.
- Ignoring brakeman warnings when using the results of `to_pdf`. The file that is generated is explicitly sent a random uuid, and it no using any user supplied data so it is a false positive flag. 
- 10-10 Health Apps
- `caregiver1010` flipper toggle

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/93581

## Testing done

- [x] *New code is covered by unit tests*

## What areas of the site does it impact?
- 10-10CG PDF download 
## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: app/controllers/v0/caregivers_assistance_claims_controller.rb [hunk 1]
```diff
@@ -39,19 +39,23 @@ def download_pdf
       # Brakeman will raise a warning if we use a claim's method or attribute in the source file name.
       # Use an arbitrary uuid for the source file name and don't use the return value of claim#to_pdf
       # as the source_file_path (to prevent changes in the the filename creating a vunerability in the future).
-      source_file_path = PdfFill::Filler.fill_form(@claim, SecureRandom.uuid, sign: false)
+      source_file_path = if Flipper.enabled?(:caregiver1010)
```
Comment (coope93) [inline app/controllers/v0/caregivers_assistance_claims_controller.rb:42]: No request specs need updated since the functionality should be exactly the same and we already have some for when the flipper toggle is enabled vs disabled.

File: app/controllers/v0/caregivers_assistance_claims_controller.rb [hunk 2]
```diff
@@ -39,19 +39,23 @@ def download_pdf
...
-      File.delete(source_file_path) if !Flipper.enabled?(:caregiver1010) && File.exist?(source_file_path)
-
       auditor.record(:pdf_download)
 
       send_data file_contents, filename: client_file_name, type: 'application/pdf', disposition: 'attachment'
     ensure
-      if Flipper.enabled?(:caregiver1010) && (source_file_path && File.exist?(source_file_path))
-        File.delete(source_file_path)
-      end
+      File.delete(source_file_path) if source_file_path && File.exist?(source_file_path)
```
Comment (coope93) [inline app/controllers/v0/caregivers_assistance_claims_controller.rb:58]: Deleting usage of the toggle for this. It has been validated in staging and is currently turned on in production. I could move this to it's own PR if it's an issue, but I thought since this change was small it would be fine.

**Review by Mitch-A6 — DISMISSED** [review]

File: app/controllers/v0/caregivers_assistance_claims_controller.rb (hunk 1 above)
Comment (coope93) [reply app/controllers/v0/caregivers_assistance_claims_controller.rb:45]: I added back the request specs with the toggle off/on since they were removed in a separate PR. They are needed again here since we are using the toggle for this logic.

File: spec/requests/v0/caregivers_assistance_claims_spec.rb [hunk 3]
```diff
@@ -189,75 +189,158 @@
       FileUtils.rm_f(response_pdf)
     end
 
-    it 'returns a completed PDF', run_at: '2017-07-25 00:00:00 -0400' do
-      expect(SavedClaim::CaregiversAssistanceClaim).to receive(:new).with(
-        form: form_data
-      ).and_return(
-        claim
-      )
+    context 'caregiver1010 toggle off' do
```
Comment (coope93) [inline spec/requests/v0/caregivers_assistance_claims_spec.rb:192]: The two context for on/off have the exact same assertions, since we expect them to be the same. The diff is larger since I broke them into two contexts.

File: spec/requests/v0/caregivers_assistance_claims_spec.rb [hunk 4]
```diff
@@ -189,75 +189,158 @@
...
+      before do
+        allow(Flipper).to receive(:enabled?).with(:caregiver1010).and_return(false)
+      end
```
Comment (stevenjcumming) [inline spec/requests/v0/caregivers_assistance_claims_spec.rb:195]: Consider this approach in the future, it's just a little easier to read:

```ruby
before do
  Flipper.enable(:caregiver1010)
end
```

**Review by stevenjcumming — APPROVED** [review]

File: spec/requests/v0/caregivers_assistance_claims_spec.rb (hunk 4 above)
Comment (Mitch-A6) [reply spec/requests/v0/caregivers_assistance_claims_spec.rb:195]: 🤔 Doesn't this leak into other tests run in the same batch?

**Comment by Mitch-A6** [comment]
side-note, but github's Hide Whitespace view was awesome for this:
Screenshot 2024-11-01 at 10 53 47 AM

File: spec/requests/v0/caregivers_assistance_claims_spec.rb (hunk 4 above)
Comment (coope93) [reply spec/requests/v0/caregivers_assistance_claims_spec.rb:195]: Yeah, unless you add 
```
after do
  Flipper.disable(:caregiver1010)
end
```
which is why I tend to use the allow instead of directly enabling/disabling. I agree it's more verbose so not as obvious, though.
